go/api_default.go

```diff
@@ -67,63 +69,99 @@ func handleNotFoundError(w http.ResponseWriter, detail string) {
 	w.Write(payload)
 }
 
-func callGrafana(url string, apiKey string, verb string, payload io.Reader) (int, []byte, error) {
+func callGrafana(admin bool, url string, auth string, verb string, payload io.Reader) (int, []byte, error) {
 	var statusCode int
 	var body []byte
 	var err error
+
+	if os.Getenv("DEBUG") == "true" {
+		logrus.Infoln("---DEBUG request logs---")
+		logrus.Infoln("URL: " + url)
+		logrus.Infoln("Method: " + verb)
+		if payload != nil {
+			buf := new(bytes.Buffer)
+			buf.ReadFrom(payload)
+			s := buf.String()
+			logrus.Infoln("Payload:" + s)
+		}
+	}
+
 	req, err := http.NewRequest(verb, url, payload)
-	req.Header.Set("Authorization", "Bearer"+" "+apiKey)
+
+	var authKey string
+	if admin {
+		if os.Getenv("DEBUG") == "true" {
+			logrus.Debugln("Using basic auth")
+		}
+		authKey = "Basic"
+	} else {
+		if os.Getenv("DEBUG") == "true" {
+			logrus.Debugln("Using bearer auth")
+		}
+		authKey = "Bearer"
+	}
+
+	req.Header.Set("Authorization", authKey+" "+auth)
 	req.Header.Set("Accept", "application/json")
 	req.Header.Set("Content-Type", "application/json")
 	client := &http.Client{}
 	resp, err := client.Do(req)
+
+	if err != nil {
+		logrus.Infoln("Error calling Grafana:", verb, url, err)
+	}
+
 	defer resp.Body.Close()
+
 	if err != nil {
-		logrus.Println("Error calling Grafana:", verb, url, err)
+		logrus.Infoln("Error calling Grafana:", verb, url, err)
 	} else {
 		body, _ = ioutil.ReadAll(resp.Body)
 		statusCode = resp.StatusCode
-		logrus.Printf("Response body from Grafana: %s", string(body))
-		logrus.Printf("Response code from Grafana: %v", statusCode)
+		if os.Getenv("DEBUG") == "true" {
+			logrus.Infof("Response body from Grafana: %s", string(body))
+			logrus.Infof("Response code from Grafana: %v", statusCode)
+		}
 	}
+
 	return statusCode, body, err
 }
 
 func getTemplateFromUrl(templateUrl string) ([]byte, error) {
 	var templateBody []byte
 	var err error
+
 	req, err := http.NewRequest("GET", templateUrl, nil)
```

Comment:
Not a biggie ... but it might be an idea just to pull the http requests into a common methods; keeps it clean and easier to log

Reply:
Sorry ... just realized you are using a common method in some places ..

Reply:
Sure makes sense.

Reply:
Yeh this is the one HTTP request which doesnt go to grafana.. maybe it can stay on its own.